src/scripts/create_hunter_csv.py

```diff
@@ -0,0 +1,477 @@
+            if col_name not in ['totalOps', 'opRate']:
+                raw_hunter_metrics_df[col_name] = raw_hunter_metrics_df[col_name].str.rstrip(
+                    ' ms')
+        # Get date based on path and add time for compatibility with hunter.
+        date_val = json_paths[0].split(os.sep)[4]
+        # As at 11pm UTC
+        date_val_w_time = f"{date_val.replace('_', '-')}{' '}{LWT_TEST_RUN_EXEC_TIME}{' +0000'}"
+
+        # Prospective case is when running hunter nightly; retrospective is when running it
+        # on already output dates-related folders of performance results (one-off analysis).
+        if is_prospective:
+            cassandra_git_short_hash, fallout_tests_git_short_hash = \
+                get_commit_hash_cass_fall_tests(is_prospective=is_prospective)
+
+        else:
+            # Swap month and day to match expected format for the git log command below
+            list_of_elems = date_val.split('_')
+            # Plus one day to then get the date's Git sha (as using 'until' in git log command)
+            date_sorted = datetime(
+                int(list_of_elems[0]), int(
+                    list_of_elems[1]), int(list_of_elems[2])
+            ) + timedelta(days=1)
+            date_sorted_y_d_m = date_sorted.strftime(FMT_Y_D_M)
+            cassandra_git_short_hash, fallout_tests_git_short_hash = \
+                get_commit_hash_cass_fall_tests(
+                    date_sorted_y_d_m, is_prospective)
+
+        # cd back into hunter_csv
+        cd_into_proj_dir()
+
+        combined_read_write_df = create_hunter_df(raw_hunter_metrics_df)
+
+        hunter_df_full = add_cols_to_metrics_df(
+            date_val_w_time,
+            cassandra_git_short_hash,
+            fallout_tests_git_short_hash,
+            combined_read_write_df
+        )
+        return hunter_df_full
+
+
+def get_hunter_df_w_test_type(json_paths: List[str], is_prospective: bool = PROSPECTIVE_MODE) -> Tuple[pd.DataFrame, str]:
+    """
+    Get the dataframe to feed to hunter with performance results and the corresponding test type (e.g., 100/1000/10000
+    partitions and either 'fixed' or 'rated') based on whether the analysis is prospective.
+
+    Args:
+        json_paths: List[str]
+                    A list of json paths with performance results and related metrics.
+        is_prospective: bool
+                    Whether the analysis is prospective (True by default).
+
+    Returns:
+            A tuple with the dataframe with performance results and the corresponding test type.
+    """
+    hunter_df_out = generate_hunter_df(json_paths, is_prospective)
+
+    if len(json_paths) == 0 or hunter_df_out.empty:
+        logging.error(
+            "Either the 'json_paths' or the hunter dataframe is empty; "
+            "thus, empty dataframe and string are being returned."
+        )
+        return pd.DataFrame(), ''
+
+    test_type = json_paths[0].split(os.sep)[5]
```

Comment:
Can you either calculate the index value or use a descriptive variable name that explains which part of the path we're indexing?

Reply:
Thanks, addressed at [a88cbe0](https://github.com/datastax/fallout-tests/pull/5/commits/a88cbe06fe955d66014ab5b995bf28e000c409ed)